vault/resource_config_ui_custom_messages.go

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/vault/api"
+)
+
+func configUICustomMessageResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: configUICustomMessageCreate,
```

Comment:
Since this endpoint is only available in Vault 1.16 we can add a minimum version requirement to the entire resource
```suggestion
		CreateContext: provider.MountCreateContextWrapper(configUICustomMessageCreate, provider.VaultVersion116),

```

Reply:
If we add this, then we are safe to merge before Vault 1.16 is released. We usually do a TFVP release shortly after the Vault release. So this will allow us to not have to scramble to get things merged and in sync between the two projects. :)